src/app/pages/questions/components/question/audio-input/audio-input.component.ts

```diff
@@ -3,112 +3,151 @@ import {
 
-// NOTE: File path to opensmile.js; Adding opensmile plugin
-import * as opensmile from '../../../../../../../plugins/cordova-plugin-opensmile/www/opensmile'
-import { Answer } from '../../../../../shared/models/answer'
+import {
+  MIN_SEC,
+  SEC_MILLISEC
+} from '../../../../../../assets/data/defaultConfig'
+import { LocKeys } from '../../../../../shared/enums/localisations'
 import { Section } from '../../../../../shared/models/question'
+import { TranslatePipe } from '../../../../../shared/pipes/translate/translate'
 import { AndroidPermissionUtility } from '../../../../../shared/utilities/android-permission'
-import { QuestionsPageComponent } from '../../../containers/questions-page.component'
+import { HomePageComponent } from '../../../../home/containers/home-page.component'
 import { AudioRecordService } from '../../../services/audio-record.service'
 
-declare var cordova: any
-declare var window: any
-
 @Component({
   selector: 'audio-input',
   templateUrl: 'audio-input.component.html'
 })
-export class AudioInputComponent implements OnInit, OnChanges {
+export class AudioInputComponent implements OnChanges, OnDestroy, OnInit {
   @Output()
   valueChange: EventEmitter<any> = new EventEmitter<any>()
   @Input()
   sections: Section[]
   @Input()
   currentlyShown: boolean
 
-  filename: string
-  name: string
-  filepath: string
-  recording: boolean
-  value: string = null
-  configFile = 'liveinput_android.conf'
-  compression = 1
-  platform = false
-  answer_b64: string = null
-  permission: any
-
-  answer: Answer = {
-    id: null,
-    value: null,
-    type: 'audio'
-  }
-
-  ngOnInit() {}
-
-  ngOnChanges() {
-    if (this.currentlyShown) {
-      this.startRecording()
-    }
-  }
+  avgWordsPerMinute = 100
+  alertShown = false
+  resumeListener
+  pauseListener
+  recordingTimeout
 
   constructor(
-    public questions: QuestionsPageComponent,
     private audioRecordService: AudioRecordService,
     private permissionUtil: AndroidPermissionUtility,
     public navCtrl: NavController,
     public alertCtrl: AlertController,
-    private device: Device
+    private platform: Platform,
+    private translate: TranslatePipe
   ) {
-    // NOTE: Stop audio recording when application is on pause / backbutton is pressed
-    document.addEventListener('pause', () => {
-      console.log('on pause')
-      // if (this.navCtrl.isActive(this.navCtrl.getActive()) == false) {
-      this.audioRecordService.stopAudioRecording()
-      // }
-    })
+    this.permissionUtil.checkPermissions()
+  }
 
-    document.addEventListener('backbutton', () => {
-      console.log('on backbutton')
-      // if (this.navCtrl.isActive(this.navCtrl.getActive()) == false) {
-      this.audioRecordService.stopAudioRecording()
-      // }
-    })
+  ngOnInit() {
+    this.onResume()
+    this.onClose()
+  }
 
-    this.permissionUtil.checkPermissions()
+  ngOnChanges() {
+    if (this.currentlyShown) this.startRecording()
```

Comment:
Why do we start recording everything there is a change? What kind of change event do we expect here?

Reply:
The change in `currentlyShown`. It will automatically start recording when the question appears.